src/startupscriptgenerator/dotnetcore/scriptgenerator.go

```diff
@@ -40,36 +40,61 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
-	scriptBuilder.WriteString("        startupDllFileNamePrefix=${file%%.runtimeconfig.json}\n")
-	scriptBuilder.WriteString("        startupDllFileName=\"$startupDllFileNamePrefix.dll\"\n")
-	scriptBuilder.WriteString("        echo \"Found the startup file '$startupDllFileName'\"\n")
-	scriptBuilder.WriteString("        break\n")
-	scriptBuilder.WriteString("      ;;\n")
-	scriptBuilder.WriteString("    esac\n")
-	scriptBuilder.WriteString("  fi\n")
-	scriptBuilder.WriteString("done\n")
 	scriptBuilder.WriteString("if [ ! -z \"$userStartUpCommand\" ]; then\n")
-	scriptBuilder.WriteString("  cd \"$appPath\"\n")
-	scriptBuilder.WriteString("  startUpCommand=\"$userStartUpCommand\"\n")
-	scriptBuilder.WriteString("  if [ -f \"$userStartUpCommand\" ]; then\n")
-	scriptBuilder.WriteString("    chmod 755 \"$userStartUpCommand\"\n")
+	scriptBuilder.WriteString("    len=${#userStartUpCommand[@]}\n")
+	scriptBuilder.WriteString("    if [ \"$len\" -eq \"1\" ]; then\n")
+	scriptBuilder.WriteString("      if [ -f \"${userStartUpCommand[$0]}\" ]; then\n")
+	scriptBuilder.WriteString("        startUpCommand=\"$userStartUpCommand\"\n")
+	scriptBuilder.WriteString("      else\n")
+	scriptBuilder.WriteString("        echo \"Could not find the startup file '${userStartUpCommand[$0]}' on disk.\"\n")
+	scriptBuilder.WriteString("      fi\n")
+	scriptBuilder.WriteString("    elif [ \"$len\" -eq \"2\" ] && [ \"${userStartUpCommand[$0]}\" == \"dotnet\" ]; then\n")
+	scriptBuilder.WriteString("      if [ -f \"${userStartUpCommand[$1]}\" ]; then\n")
+	scriptBuilder.WriteString("        startUpCommand=\"$userStartUpCommand\"\n")
+	scriptBuilder.WriteString("      else\n")
+	scriptBuilder.WriteString("        echo \"Could not find the file '${userStartUpCommand[$1]}' on disk.\"\n")
+	scriptBuilder.WriteString("      fi\n")
+	scriptBuilder.WriteString("    else\n")
+	scriptBuilder.WriteString("      startUpCommand=\"$userStartUpCommand\"\n")
+	scriptBuilder.WriteString("    fi\n")
+	scriptBuilder.WriteString("fi\n\n")
+
+	scriptBuilder.WriteString("if [ -z \"$startUpCommand\" ]; then\n")
+	scriptBuilder.WriteString("  echo Finding the startup file name...\n")
+	scriptBuilder.WriteString("  for file in *; do \n")
+	scriptBuilder.WriteString("    if [ -f \"$file\" ]; then \n")
+	scriptBuilder.WriteString("      case $file in\n")
+	scriptBuilder.WriteString("        *.runtimeconfig.json)\n")
+	scriptBuilder.WriteString("          startupDllFileNamePrefix=${file%%.runtimeconfig.json}\n")
+	scriptBuilder.WriteString("          startupExecutableFileName=\"$startupDllFileNamePrefix\"\n")
+	scriptBuilder.WriteString("          startupDllFileName=\"$startupDllFileNamePrefix.dll\"\n")
+	scriptBuilder.WriteString("          break\n")
+	scriptBuilder.WriteString("        ;;\n")
+	scriptBuilder.WriteString("      esac\n")
+	scriptBuilder.WriteString("    fi\n")
+	scriptBuilder.WriteString("  done\n\n")
+	scriptBuilder.WriteString("  if [ -f \"$startupExecutableFileName\" ]; then\n")
```

Comment:
It just occurred to me that if the app was built for, say, Mac, this won't work. But the existing code will, since it will run the app using `dotnet app.dll`. I think we should leave this for the next iteration so we have time to think about this a bit more.

Reply:
Are you referring to self-contained app? IMO that would be a user error as the user is clearly deploying the app to Linux containers. Like one has to explicitly use osx-x64 or linux-x64 for example. Thoughts?

Reply:
It's right that the app was built as self contained for one platform and deployed to another, but an app built this way currently _runs_ (we we'll see the `.dll`file  and run it with `dotnet`), but without any user action after this update it will stop working. 

So from the user perspective, they had something working that now doesn't and they will be caught by surprise. We could check the file type and execution permission, for example.

Reply:
I tried the following scenario now - publish the app for Mac and do a `dotnet foo.dll` on linux...
and this failed to execute...so we need not worry about this scenario as it would never have worked.

FYI:
**Publish for Mac but run on Linux**
![image](https://user-images.githubusercontent.com/6559784/57771910-f9eb6080-76c8-11e9-992f-6d9bd2d9dadb.png)

**Publish for Linux and run on Linux**
![image](https://user-images.githubusercontent.com/6559784/57771972-256e4b00-76c9-11e9-83c3-3273a95db973.png)